executable/dos_mz.ksy

```diff
@@ -21,22 +22,32 @@ doc: |
   segment of raw CPU instructions), DOS MZ .exe file format allowed
   more flexible memory management, loading of larger programs and
   added support for relocations.
+doc-ref:
+  - http://www.delorie.com/djgpp/doc/exe/
 seq:
-  - id: hdr
-    type: mz_header
-  - id: mz_header2
-    size: hdr.ofs_relocations - 0x1c
-  - id: relocations
-    type: relocation
-    repeat: expr
-    repeat-expr: hdr.num_relocations
+  - id: header
+    type: exe_header
   - id: body
     size-eos: true
+instances:
+  relocations:
+    pos: header.mz.ofs_relocations
+    io: header._io
+    type: relocation
+    repeat: expr
+    repeat-expr: header.mz.num_relocations
+    if: header.mz.ofs_relocations != 0
 types:
+  exe_header:
+    seq:
+      - id: mz
+        type: mz_header
+      - id: rest_of_header
+        size: mz.len_header - mz._sizeof
   mz_header:
     seq:
       - id: magic
-        size: 2
+        contents: 'MZ'
```

Comment:
I think the magic number can also be `ZM` instead of `MZ`, at least in some cases (DOS but not Windows, possibly?). I don't have a good source for this sadly and don't remember where I first read it. But if you search for 'exe magic number "zm"' with your favorite search engine, you can find a number of people mentioning this.

Reply:
There are a few sources that say it, but it is apparently very rare. I will add it.